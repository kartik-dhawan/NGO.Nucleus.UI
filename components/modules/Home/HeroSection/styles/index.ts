export const styles = {
  heroSectionWrapper: {
    display: "flex",
    flexDirection: "column",
    gap: {
      xs: "2rem",
      md: "12rem",
    },
  },
  heroSectionItemWrapper: {
    display: "flex",
    gap: {
      sm: "2rem",
      lg: "10rem",
    },
    position: "relative",
  },
  heroSectionTypographyWrapper: {
    flex: 4,
    padding: "24px 32px",
    boxSizing: "border-box",
  },
  heroSectionTypography: {
    position: {
      md: "sticky",
    },
    top: "78px",
    "& > h1": {
      margin: "4px 0px",
      fontSize: {
        xs: "32px",
        xl: "38px",
      },
    },
    "& p": {
      fontSize: {
        xs: "18px",
        xl: "21px",
      },
      overflow: "hidden",
      textOverflow: "ellipsis",
      display: "-webkit-box",
      WebkitLineClamp: {
        md: 5,
        lg: 7,
      },
      WebkitBoxOrient: "vertical",
    },
  },
  heroSectionImages: {
    flex: 6,
    "& > img": {
      width: "100%",
      height: {
        xs: "100vh",
        md: "100%",
      },
      objectFit: "cover",
      objectPosition: "50% 30%",
    },
  },
  heroLandingSectionButton: {
    fontSize: "14px",
    fontWeight: "bold",
    color: "#d9d9d9",
    backgroundColor: "#111",
    borderRadius: "10000px",
    padding: "12px 20px 12px 24px",
    margin: "12px 0px",
    textTransform: "capitalize",
    "& > svg": {
      transition: "550ms all ease",
      marginLeft: "12px",
    },
    "&:hover": {
      "& > svg": {
        transform: "translateX(12px)",
      },
    },
  },
}
