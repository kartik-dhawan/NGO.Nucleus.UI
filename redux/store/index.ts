import { configureStore } from "@reduxjs/toolkit"
import contentSlice from "../slices/contentSlice"

const store = configureStore({
  reducer: {
    contentSlice,
  },
})

export type RootType = ReturnType<typeof store.getState>

export default store
