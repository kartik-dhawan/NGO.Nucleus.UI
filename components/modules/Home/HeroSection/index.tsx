import { useSelector } from "react-redux"
import { RootType } from "../../../../redux/store"
import { Box, Typography } from "@mui/material"
import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
import Image from "next/image"

const HeroSection = () => {
  const { content } = useSelector((state: RootType) => state.contentSlice)

  return (
    <Box
      component="section"
      sx={{ display: "flex", flexDirection: "column", gap: "10rem" }}
    >
      {content.heroSectionImages &&
        content.heroSectionImages.map((item: any, index: number) => {
          return (
            <Box
              key={item.sys.id}
              sx={{
                display: "flex",
                gap: "5rem",
                flexDirection: index === 0 ? "row" : "row-reverse",
                position: "relative",
              }}
            >
              <Box
                sx={{
                  flex: 4,
                  padding: "24px 32px",
                  boxSizing: "border-box",
                }}
              >
                <Typography
                  className="barlowCondensed"
                  sx={{
                    position: "sticky",
                    top: "78px",
                    // left: 0,
                    "& > h1": {
                      margin: "4px 0px",
                      fontSize: "32px",
                    },
                    "& p": {
                      fontSize: "18px",
                    },
                  }}
                >
                  {index === 0 &&
                    documentToReactComponents(content?.heroLandingSectionBody)}
                  {index === 1 &&
                    documentToReactComponents(content?.heroQuoteSectionBody)}
                </Typography>
                <Box></Box>
              </Box>
              <Box
                sx={{
                  flex: 5,
                  "& > img": {
                    width: "100%",
                    height: "100%",
                  },
                }}
              >
                {content?.heroSectionImages && (
                  <Image
                    src={
                      "https:" +
                      content?.heroSectionImages[index].fields?.file.url
                    }
                    alt={content?.heroSectionImages[index].fields.description}
                    width={2787}
                    height={4180}
                  ></Image>
                )}
              </Box>
            </Box>
          )
        })}
    </Box>
  )
}

export default HeroSection
