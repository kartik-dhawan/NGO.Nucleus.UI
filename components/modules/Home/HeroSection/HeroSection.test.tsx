import renderer from "react-test-renderer"
import { Provider } from "react-redux"
import "@testing-library/jest-dom"
import store from "../../../../redux/store"
import HeroSection from "."

describe("Hero section", () => {
  it("-- should render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HeroSection />
        </Provider>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
