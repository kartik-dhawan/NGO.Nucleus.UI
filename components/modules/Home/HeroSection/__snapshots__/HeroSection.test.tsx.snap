// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero section -- should render correctly 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-iat75x{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}@media (min-width:0px){.css-iat75x{gap:2rem;}}@media (min-width:900px){.css-iat75x{gap:12rem;}}",
      }
    }
    data-emotion="css iat75x"
  />,
  <section
    className="MuiBox-root css-iat75x"
  />,
]
`;
